### PR TITLE
mrc-2061 Front end for async calibrate

### PR DESCRIPTION
**Pull request by EmmaLRussell**
## Description

Add in the front end actions and mutations to execute model calibration asynchronously. The actions co-ordinate each other - a successful `submit` action dispatches an action to start polling for status, and when a status result comes back indicating successful completion, the `getResult` action is dispatched. Hence all the component needs to do is dispatch the initial submit action. 

This branch also moves using the barchart metadata from the model run result (it's no longer returned there) to using the metadata from the calibrate result. 

## Type of version change
_Delete as appropriate_
Patches

## Checklist

- [x] I have incremented version number, or version needs no increment
- [x] The build passed successfully, or failed because of ADR tests

### CODE REVIEW

File: src/app/static/src/app/store/modelCalibrate/actions.ts [hunk 1]
```diff
@@ -21,29 +24,82 @@ export const actions: ActionTree<ModelCalibrateState, RootState> & ModelCalibrat
         const response = await api<ModelCalibrateMutation, ModelCalibrateMutation>(context)
             .withSuccess(ModelCalibrateMutation.ModelCalibrateOptionsFetched)
             .ignoreErrors()
-            .get<DynamicFormMeta>("/model/calibrate/options/");
+            .get<DynamicFormMeta>("model/calibrate/options/");
```
Comment (Nicolas-Dolan) [inline src/app/static/src/app/store/modelCalibrate/actions.ts:27]: Is the extra slash just good practice or was it needed to run the code?

File: src/app/static/src/tests/components/modelCalibrate/modelCalibrate.test.ts [hunk 2]
```diff
@@ -13,7 +13,7 @@ import ErrorAlert from "../../../app/components/ErrorAlert.vue";
 import {ModelCalibrateMutation} from "../../../app/store/modelCalibrate/mutations";
 
 describe("Model calibrate component", () => {
-    const getStore = (state: Partial<ModelCalibrateState> = {}, fetchAction = jest.fn(), calibrateAction = jest.fn(),
+    const getStore = (state: Partial<ModelCalibrateState> = {}, fetchAction = jest.fn(), submitction = jest.fn(),
```
Comment (Nicolas-Dolan) [inline src/app/static/src/tests/components/modelCalibrate/modelCalibrate.test.ts:16]: typo: `submitAction`

File: src/app/static/src/tests/components/modelCalibrate/modelCalibrate.test.ts [hunk 3]
```diff
@@ -23,7 +23,7 @@ describe("Model calibrate component", () => {
                     state: mockModelCalibrateState(state),
                     actions: {
                         fetchModelCalibrateOptions: fetchAction,
-                        calibrate: calibrateAction
+                        submit: submitction
```
Comment (Nicolas-Dolan) [inline src/app/static/src/tests/components/modelCalibrate/modelCalibrate.test.ts:26]: again here: `submitAction`

**Review by Nicolas-Dolan — APPROVED** [review]
Changes make sense and it works well when I ran it locally.

File: src/app/static/src/app/store/modelCalibrate/actions.ts (hunk 1 above)
Comment (EmmaLRussell) [reply src/app/static/src/app/store/modelCalibrate/actions.ts:27]: I think this was needed - bug in previous branch

**Pull request merged**